rest/src/main/kotlin/builder/interaction/ApplicationCommandBuilders.kt

```diff
@@ -2,83 +2,59 @@ package dev.kord.rest.builder.interaction
-import dev.kord.common.entity.PartialDiscordGuildApplicationCommandPermissions
-import dev.kord.common.entity.Snowflake
+import dev.kord.common.entity.Permissions
 import dev.kord.rest.builder.RequestBuilder
 import dev.kord.rest.json.request.ApplicationCommandCreateRequest
 import dev.kord.rest.json.request.ApplicationCommandModifyRequest
-import dev.kord.rest.json.request.ApplicationCommandPermissionsEditRequest
-import kotlin.contracts.InvocationKind
-import kotlin.contracts.contract
 
 @KordDsl
 public interface ApplicationCommandCreateBuilder : LocalizedNameCreateBuilder,
     RequestBuilder<ApplicationCommandCreateRequest> {
+
+    public var defaultMemberPermissions: Permissions?
+
+    @Deprecated("'defaultPermission' is deprecated in favor of 'defaultMemberPermissions' and 'dmPermission'. Setting 'defaultPermission' to false can be replaced by setting 'defaultMemberPermissions' to empty Permissions and 'dmPermission' to false ('dmPermission' is only available for global commands).")
     public var defaultPermission: Boolean?
     public val type: ApplicationCommandType
-}
 
-@KordDsl
-public interface ApplicationCommandModifyBuilder : LocalizedNameModifyBuilder,
-    RequestBuilder<ApplicationCommandModifyRequest> {
-    public var defaultPermission: Boolean?
+    /**
+     * Disables the command for everyone by default.
+     *
+     * **This does not ensure normal users cannot execute the command, any server owner can change this setting**
+     */
+    public fun disableCommand() {
+        defaultMemberPermissions = Permissions()
+    }
 }
 
 @KordDsl
-public class ApplicationCommandPermissionsBulkModifyBuilder :
-    RequestBuilder<List<PartialDiscordGuildApplicationCommandPermissions>> {
-
-    @PublishedApi
-    internal val permissions: MutableMap<Snowflake, ApplicationCommandPermissionsModifyBuilder> = mutableMapOf()
-
-    public inline fun command(
-        commandId: Snowflake,
-        builder: ApplicationCommandPermissionsModifyBuilder.() -> Unit,
-    ) {
-        contract {
-            callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
-        }
-
-        permissions[commandId] = ApplicationCommandPermissionsModifyBuilder().apply(builder)
-    }
-
-    override fun toRequest(): List<PartialDiscordGuildApplicationCommandPermissions> {
-        return permissions.map { (id, builder) ->
-            PartialDiscordGuildApplicationCommandPermissions(
-                id, builder.permissions.toList()
-            )
-        }
+public interface GlobalApplicationCommandCreateBuilder : ApplicationCommandCreateBuilder,
+    RequestBuilder<ApplicationCommandCreateRequest> {
+    public var dmPermission: Boolean?
+
+    /**
+     * Disables the command for everyone by default.
+     *
+     * **This only ensures that DM users cannot execute the command, any server owner can change this setting**
+     */
+    override fun disableCommand() {
+        super.disableCommand()
+        dmPermission = false
```

Comment:
because of the virtual override and the only implementation implementing the global builder, this will always set `dmPermission` to `false` (also for guild commands)

Reply:
Does this really matter? won't Discord just ignore this for guilds? Like this we would need to introduce this parameter to all builders

Reply:
I haven't tested it, so I don't know if Discord will ignore it, but I would prefer to do this properly.

Reply:
Either we ignore it, we just don't have this function at all because it's less work to just do this manually, or we just drop dm support in the function

Reply:
> or we just drop dm support in the function

I think this is the easiest thing to do, if we also rename it to `disableCommandInGuild()` it should be clear that it doesn't do anything about dms

Another note: the docs of this method should refer to admins not server owners.